### PR TITLE
[CORE] Fix format and add UnitTest of class RoundRobinPartitioner

### PR DESCRIPTION
**Pull request by kecookier**
## What changes were proposed in this pull request?

(Please fill in changes proposed in this fix)

(Fixes: \#ISSUE-ID)

## How was this patch tested?

(Please explain how this patch was tested. E.g. unit tests, integration tests, manual tests)

(If this patch involves UI changes, please attach a screenshot; otherwise, remove this)

### CODE REVIEW

**Comment by kecookier** [comment]
@zuochunwei  Please help review.

File: cpp/core/tests/RoundRobinPartitionerTest.cc [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+    checkResult(row2Part, part2RowCount);
+  }
+  {
+    int numRows = 10;
+    ASSERT_TRUE(partitioner_->compute(nullptr, numRows, row2Partition_, partition2RowCount_).ok());
+    ASSERT_EQ(getPidSelection(), 8);
+    std::vector<uint16_t> row2Part(numRows);
+    std::generate_n(row2Part.begin(), numRows, [n = 8, numPart]() mutable { return (n++) % numPart; });
+    checkResult(row2Part, std::vector<uint32_t>(numPart, 1));
+  }
```
Comment (zuochunwei) [inline cpp/core/tests/RoundRobinPartitionerTest.cc:141]: please add a spaceline between `}` and `{`

File: cpp/core/tests/RoundRobinPartitionerTest.cc [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+  {
+    int numRows = 12;
+    ASSERT_TRUE(partitioner_->compute(nullptr, numRows, row2Partition_, partition2RowCount_).ok());
+    ASSERT_EQ(getPidSelection(), 0);
+    std::vector<uint16_t> row2Part(numRows);
+    std::generate_n(row2Part.begin(), numRows, [n = 8, numPart]() mutable { return (n++) % numPart; });
+    std::vector<uint32_t> part2RowCount(numPart, 1);
+    std::fill_n(part2RowCount.begin() + 8, numRows - numPart, 2);
+    checkResult(row2Part, part2RowCount);
+  }
```
Comment (zuochunwei) [inline cpp/core/tests/RoundRobinPartitionerTest.cc:151]: the same to above comment

File: cpp/core/tests/RoundRobinPartitionerTest.cc [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+#include <numeric>
+namespace gluten {
+class RoundRobinPartitionerTest : public ::testing::Test {
+ protected:
+  void prepareData(int numPart) {
+    partitioner_ = ShuffleWriter::Partitioner::create<RoundRobinPartitioner>(numPart, false);
+    row2Partition_.clear();
+    partition2RowCount_.clear();
+    partition2RowCount_.resize(numPart);
+  }
```
Comment (zuochunwei) [inline cpp/core/tests/RoundRobinPartitionerTest.cc:29]: add a new spaceline between the two functions' definitions.

File: cpp/core/tests/RoundRobinPartitionerTest.cc [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "shuffle/RoundRobinPartitioner.h"
+#include <gtest/gtest.h>
+#include <cstdlib>
+#include <numeric>
+namespace gluten {
```
Comment (zuochunwei) [inline cpp/core/tests/RoundRobinPartitionerTest.cc:21]: add a new spaceline here

File: cpp/core/tests/RoundRobinPartitionerTest.cc [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+    int numRows = 22;
+    ASSERT_TRUE(partitioner_->compute(nullptr, numRows, row2Partition_, partition2RowCount_).ok());
+    ASSERT_EQ(getPidSelection(), 2);
+    std::vector<uint16_t> row2Part(numRows);
+    std::generate_n(row2Part.begin(), numRows, [n = 0, numPart]() mutable { return (n++) % numPart; });
+    std::vector<uint32_t> part2RowCount(numPart, 2);
+    std::fill_n(part2RowCount.begin(), numRows - 2 * numPart, 3);
+    checkResult(row2Part, part2RowCount);
+  }
+}
```
Comment (zuochunwei) [inline cpp/core/tests/RoundRobinPartitionerTest.cc:162]: need a new spaceline here

**Review by zuochunwei — APPROVED** [review]

**Pull request merged**